File: c/grains/src/grains.c
#include "grains.h"

#define MIN 1
#define MAX 64

uint64_t square(uint8_t index) {

    if (index < MIN || index > MAX) {
        return 0;
    }

    return powl(2, index - 1);
}


uint64_t total(void) {
    return powl(2, 64) - 1;
}

File: python/grains/grains.py
__MIN = 1
__MAX = 64
__TOTAL = (2 << 63) - 1


def square(number):

    if number < __MIN or number > __MAX:
        raise ValueError("Value out of range")

    return 2 ** (number - 1)


def total():
    return __TOTAL
